### DISCUSSION TITLE
主应用和子应用都基于umi开发，关于部署打包  P_XXXX.js页面在集成的环境和单独子应用加载路径不同的问题

### DISCUSSION THREAD

**Discussion by dongcer** [umijs/qiankun#1651] [Q&A]
localhost:7000 主应用  7001子应用 

nginx代理配置
server {
        listen       7000;
        add_header Access-Control-Allow-Origin *;
        add_header Access-Control-Allow-Methods 'GET, POST, OPTIONS';
        add_header Access-Control-Allow-Headers 'DNT,X-Mx-ReqToken,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization';

        if ($request_method = 'OPTIONS') {
           return 204;
        }
        server_name  localhost;

        location / {
             root   d:/www/main;
             index  index.html index.htm;
             try_files $uri $uri/ /index.html;
        }
        location /app1/ {
            add_header Access-Control-Allow-Origin *;
            add_header Access-Control-Allow-Methods 'GET, POST, OPTIONS';
            add_header Access-Control-Allow-Headers 'DNT,X-Mx-ReqToken,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization';

            proxy_pass http://127.0.0.1:7001/;
         }
        error_page   500 502 503 504  /50x.html;
        location = /50x.html {
            root   html;
        }
}
server {
        listen       7001;
        add_header Access-Control-Allow-Origin *;
        add_header Access-Control-Allow-Methods 'GET, POST, OPTIONS';
        add_header Access-Control-Allow-Headers 'DNT,X-Mx-ReqToken,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Authorization';

        if ($request_method = 'OPTIONS') {
            return 204;
        }
        server_name  localhost;

        location / {
            root   d:/www/app1;
            index  index.html index.htm;
            try_files $uri $uri/ /index.html;
        }
        error_page   500 502 503 504  /50x.html;
        location = /50x.html {
            root   html;
        }
}

出现的问题：
子应用： publicPath: `http://localhost:7000/app1/`,
单独打开可以正常访问  
![image](https://user-images.githubusercontent.com/35223950/129435966-9967f84f-e03f-4d6c-a9cd-1f4022f9df4a.png)

但集成到主应用里：
![image](https://user-images.githubusercontent.com/35223950/129435985-e098d1e5-98af-481f-809b-e470708b6af3.png)
![image](https://user-images.githubusercontent.com/35223950/129435998-8bf5430f-5826-4ad2-a94f-c7e98116b11f.png)
这个子应用页面 p_keepalive.js 路径上就没有/app1/  了，导致只显示了umi的loading...页面
![image](https://user-images.githubusercontent.com/35223950/129436031-984ad365-42bc-4553-886b-c77b65808a4b.png)

请问有没有人能帮助一下

**Comment by Dashuai-Black** [comment]
哈喽 请问遇到过 主应用加载子应用  路由跳转的时候 加载承载子应用节点的顺序问题么，大概意思就是子应用返回找节点的时候节点dom还没有渲染完  导致渲染失败 ，这个问题出现的概率还没有找到 有时候能加载出来子应用 有时候就失败 F`12报的错误就是没找到dom节点
